azure/services/scalesets/scalesets.go

```diff
@@ -364,6 +365,7 @@ func (s *Service) buildVMSSFromSpec(ctx context.Context, vmssSpec azure.ScaleSet
 			Tier:     to.StringPtr("Standard"),
 			Capacity: to.Int64Ptr(vmssSpec.Capacity),
 		},
+		Zones: to.StringSlicePtr(vmssSpec.FailureDomains),
```

Comment:
It might be nice to validate that the zones are valid in that region for that VM size. We can leverage the SKU cache for this, similar to how we check for VM SKU capabilities like ephemeral OS disk support and number of vCPUs

Reply:
How we decide when to validate in the webhook vs in the reconciler? Validating the zones in the webhook would be a better UX I think, because the user would get feedback right away.

Reply:
Whenever we can we do validation in the webhooks but for anything that requires calling out to Azure (and SKU validation is one of them since it varies per sub and evolves over time) we do it in the reconciler

Agree the UX is better in the webhook but if you have to wait 2 minutes for the webhook to get a response from Azure APIs it becomes pretty bad UX

Reply:
Added AZ validation.